nixos/modules/programs/goldwarden.nix

```
@@ -0,0 +1,27 @@
+{ lib, config, pkgs, ... }:
+let
+  cfg = config.programs.goldwarden;
+in
+{
+  options.programs.goldwarden = {
+    enable = lib.mkEnableOption "Goldwarden";
+    package = lib.mkPackageOption pkgs "goldwarden" {};
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [
+      # for cli and polkit action
+      cfg.package
+      # binary exec's into pinentry which should match the DE
+      pkgs."pinentry-${config.programs.gnupg.agent.pinentryFlavor}"
+    ];
+
+    # see https://github.com/quexten/goldwarden/blob/main/cmd/setup_linux.go
+    systemd.user.services.goldwarden = {
+      description = "Goldwarden daemon";
+      wantedBy = [ "graphical-session.target" ];
+      after = [ "graphical-session.target" ];
```

Inspired by the ssh-agent service https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/programs/ssh.nix#L327. This way this service is only run for non-system user accouts.
```suggestion
      after = [ "graphical-session.target" ];
      unitConfig.ConditionUser = "!@system";
```